### ISSUE TITLE
Add 'with_fixed_indentation' flag to the `Layout/ArrayAlignment` cop

### ISSUE THREAD

**Issue by nikitasakov** [rubocop-hq/rubocop#7654]
## Is your feature request related to a problem? Please describe.

We in our company decided to use `Layout/ArgumentAlignment` and `Layout/ParameterAlignment` with `with_fixed_indentation`. Also we would like to use this flag for `Layout/ArrayAlignment` too, but this cop doesn't support `with_fixed_indentation`.

## Describe the solution you'd like

I think it would be cool to add `with_fixed_indentation` flag for `Layout/ArrayAlignment` as for `Layout/ArgumentAlignment` and `Layout/ParameterAlignment`.

## Describe alternatives you've considered

I don't see any  alternatives

## Additional context

We would like to write arrays this way:
```ruby
array = [1, 2, 3,
  4, 5, 6]
```
